src/engraving/libmscore/note.cpp

```diff
@@ -2354,6 +2363,10 @@ void Note::updateAccidental(AccidentalState* as)
             } else if (acci == AccidentalType::NONE) {
                 acci = AccidentalType::NATURAL;
             }
+        } else if (chordHasAnotherAccidentalForSameNote(this)) {
```

Comment:
potentially this condition could be just be added to the `if` above, but hard to be confident that won't have other consequences